### PR TITLE
feat: add global.imagePullSecrets parameter for private registry authentication

### PR DESCRIPTION
**Pull request by ytimocin**
- Add global.imagePullSecrets to values.yaml and all deployment/statefulset templates
- Update CLI examples in rad init, install, and upgrade commands
- Add comprehensive Helm unit tests for multiple pull secrets
- Update Chart README with configuration examples

# Description

This pull request adds comprehensive support for configuring image pull secrets across all Radius Helm chart deployments, enabling seamless use of private container registries that require authentication. It includes updates to documentation, Helm templates, default values, CLI usage examples, and Helm chart tests to ensure correct handling and validation of `imagePullSecrets`.

**Enhancements for private registry support:**

* Added `global.imagePullSecrets` configuration to `values.yaml`, allowing users to specify one or more Kubernetes secrets for pulling images from private registries.
* Updated deployment and statefulset templates (`controller`, `dashboard`, `database`, `de`, `dynamic-rp`, `rp`, `ucp`) to inject `imagePullSecrets` when specified, ensuring all components can authenticate to private registries. [[1]](diffhunk://#diff-6e17741710603ff5eab2c2df3885c905ebc5c657cf16232734059f645f97397fR30-R33) [[2]](diffhunk://#diff-fa8b3636d5cfdf8b941b89a5af4b2f3725e7ddc086f406111a08afbe14ce4267R25-R28) [[3]](diffhunk://#diff-9366e0a3212639c13de30e1ab3896ba45398986535acd618702b94c38ecab701R25-R28) [[4]](diffhunk://#diff-aa649ac3a27b3a57cd4da52d741118f2870fd721886d603f5091a90fc825e025R33-R36) [[5]](diffhunk://#diff-bb1c415d2aa24ec5f2f090395448250831b3b467289e1fa6c603fd1661453de1R33-R36) [[6]](diffhunk://#diff-92bd08c7c2302e12d2ceb2f7bf0a5bdd0016bb8e2eb2639588e994366e9ee841R33-R36) [[7]](diffhunk://#diff-0fe5610e45572cab258709e7346308d2bdc2f9c16a46d493f049de4707ea465dR33-R36)

**Documentation and usage improvements:**

* Expanded documentation in `README.md` with step-by-step instructions for using private registries and configuring image pull secrets, including both Helm and `rad` CLI workflows.
* Added CLI usage examples for `rad install`, `rad upgrade`, and `rad init` commands to demonstrate how to set `imagePullSecrets` for single or multiple registries. [[1]](diffhunk://#diff-969a1495c18d847fc09b31b1e6854e7be42d541513807361aaa5e9497a62e86fR71-R78) [[2]](diffhunk://#diff-5a9f77a14a8179c704434f4c6c41708529bb7bf86eddc14759fedb66def1dc4fR78-R85) [[3]](diffhunk://#diff-043b8a9fb42a3b3d2876c07a59c8975147156ed05f3e2c30c934e8a0e65d2ecfR87-R94)

**Testing and validation:**

* Introduced new Helm chart tests to validate correct rendering and propagation of `imagePullSecrets` across all relevant deployments, including support for multiple secrets and various configuration scenarios.

**Miscellaneous:**

* Minor update to `go.mod` to remove the `// indirect` comment from the `golang.org/x/mod` dependency, with no functional impact.

## Type of change

- This pull request adds or changes features of Radius and has an approved issue (issue link required).

Fixes: #issue_number

## Contributor checklist
Please verify that the PR meets the following requirements, where applicable:

- An overview of proposed schema changes is included in a linked GitHub issue.
    - [ ] Yes 
    - [x] Not applicable 
- A design document PR is created in the [design-notes repository](https://github.com/radius-project/design-notes/), if new APIs are being introduced.
    - [ ] Yes 
    - [x] Not applicable 
- The design document has been reviewed and approved by Radius maintainers/approvers.
    - [ ] Yes 
    - [x] Not applicable 
- A PR for the [samples repository](https://github.com/radius-project/samples) is created, if existing samples are affected by the changes in this PR.
    - [ ] Yes 
    - [x] Not applicable 
- A PR for the [documentation repository](https://github.com/radius-project/docs) is created, if the changes in this PR affect the documentation or any user facing updates are made.
    - [ ] Yes 
    - [x] Not applicable 
- A PR for the [recipes repository](https://github.com/radius-project/recipes) is created, if existing recipes are affected by the changes in this PR.
    - [ ] Yes 
    - [x] Not applicable

### CODE REVIEW

File: deploy/Chart/README.md [hunk 1]
```diff
@@ -104,12 +104,51 @@ helm upgrade --wait --install radius deploy/Chart -n radius-system \
   --set-file global.rootCA.cert=/path/to/ca-certificate.pem
 ```
 
+#### Using with Private Registries and Authentication
+
+For private registries that require authentication:
+
+1. Create the docker-registry secret in the namespace:
+
```
Comment (lakshmimsft) [inline deploy/Chart/README.md:112]: nit: ... in the `radius-system` namespace

File: deploy/Chart/tests/helpers_test.yaml [hunk 2]
```diff
@@ -105,6 +105,114 @@ tests:
...
+      # Verify it applies to other deployments too
+      - equal:
+          path: spec.template.spec.containers[0].image
+          value: private.registry.io/applications-rp:0.48
+        template: rp/deployment.yaml
+      - equal:
+          path: spec.template.spec.imagePullSecrets[0].name
+          value: azure-cred
+        template: rp/deployment.yaml
+
```
Comment (lakshmimsft) [inline deploy/Chart/tests/helpers_test.yaml:215]: Should we add negative tests for malformed secret objects, missing name field, empty lists.

Comment (lakshmimsft) [inline deploy/Chart/tests/helpers_test.yaml:215]: Another suggestion is adding file values.schema.json file to fail fast if 'name' does not exist in imagePullSecrets:
```
{
  "$schema": "https://json-schema.org/draft-07/schema#",
  "type": "object",
  "properties": {
    "global": {
      "type": "object",
      "properties": {
        "imagePullSecrets": {
          "type": "array",
          "items": {
            "type": "object",
            "properties": {
              "name": { "type": "string", "minLength": 1 }
            },
            "required": ["name"],
            "additionalProperties": false
          }
        }
      },
      "additionalProperties": true
    }
  },
  "additionalProperties": true
}
```

Comment (ytimocin) [reply deploy/Chart/tests/helpers_test.yaml:215]: Good idea but we have to do this for all of the values.yaml. Created this follow-up issue for this: https://github.com/radius-project/radius/issues/10186.

Comment (ytimocin) [reply deploy/Chart/tests/helpers_test.yaml:215]: Added some negative tests.

**Review by lakshmimsft — APPROVED** [review]
